woocommerce/wc-admin
# D3 Chart: legend component and integration

This PR aims to introduce the legend components and handle their integration with the chart. The master thread for the [D3 Chart Component can be found here](https://github.com/woocommerce/wc-admin/issues/164).

This PR includes:
- a refactor of the D3 base component
- core legend layout and interactivity with the D3 chart
- folder restructure
- (checkboxes and methods for testing interactivity)
- initial legend README section

There will be more changes coming to the `<Legend />` component, including styling and potentially adding the legend as a prop to the chart (similar to the tooltip).

### Testing:
- Toggle through the basic checkboxes to ensure that the legend moves to the left of the chart when there are more than 2 categories/products and on the top when there are 1 or 2 categories/products.
- Hover over the legend items to ensure that they are brought into focus and the others fade out a bit
- Click the legend items (boxes and text) to ensure that they are included and excluded as you toggle them on and off respectively
![legend](https://user-images.githubusercontent.com/1160732/43076304-1c0b03b0-8e84-11e8-8875-4d8033b3b9ce.gif)
